java/src/shared/models/game/GameMap.java

```diff
@@ -77,6 +80,118 @@ public GameMap(Map<EdgeLocation, PlayerIndex> roads,
         this.cities = cities;
     }
 
+    /**
+     * Constructor that creates GameMap based on if the user wants random hex types, random port types,
+     * and/or random numbers on hexes
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomPorts   whether the Port Types are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public GameMap(boolean randomTiles, boolean randomPorts, boolean randomNumbers) {
+        setNewGameHexes(randomTiles, randomNumbers);
+        setNewGamePorts(randomPorts);
+    }
+
+    /**
+     * Sets the hexes of the new game based on if the user wants random hex types or random numbers on hexes
+     *
+     * @param randomTiles   whether the HexTypes are random
+     * @param randomNumbers whether the Numbers on the Hexes are random
+     */
+    public void setNewGameHexes(boolean randomTiles, boolean randomNumbers) {
+        List<Integer> hexNumbers = setHexNumbers(randomNumbers);
+
+        List<HexType> hexTypes = setHexTypes(randomTiles);
+
+        ArrayList<HexLocation> hexLocations = setHexLocations();
+
+        for (int i = 0; i < hexLocations.size(); i++) {
+            hexes.put(hexLocations.get(i), new Hex(hexNumbers.get(i), hexLocations.get(i), hexTypes.get(i)));
+        }
+    }
+
+    /**
+     * Sets the hex locations of the new game based on the radius
+     *
+     * @return the List of Hex Locations for each hex on the map
+     */
+    public ArrayList<HexLocation> setHexLocations() {
+        ArrayList<HexLocation> hexLocations = new ArrayList<>();
+
+        for (int column = -radius + 1; column < radius; column++) {
+            for (int diagonalRow = (column < 0) ? -radius - column + 1 : -radius + 1; diagonalRow < ((column > 0) ? radius - column : radius); diagonalRow++) {
+                hexLocations.add(new HexLocation(column, diagonalRow));
+            }
+        }
+        return hexLocations;
+    }
+
+    /**
+     * Sets the Hex Types for each hex on the new game map
+     * @param randomTiles whether the HexTypes are random
+     * @return List of Hex Types that contain a hex type for each hex on the new game map
+     */
+    public List<HexType> setHexTypes(boolean randomTiles) {
```

Comment:
Will it (or should it) ever be used outside of the own class? If the answer is no, then make it private.